client/components/RewardTier.jsx

```diff
@@ -1,22 +1,134 @@
 import React from 'react';
+import styled from 'styled-components';
+import MiniPledgeForm from './MiniPledgeForm.jsx';
 
-const RewardTier = (props) => (
-  <div className='rewardTier'>
-    <div className='pledgeAmount'>Pledge US$ {props.reward.pledgeAmount} or more</div>
-    <div className='rewardName'>{props.reward.name}</div>
-    <div className='rewardDesc'>{props.reward.description}</div>
-    <div className='rewardItems'>
-      INCLUDES:
-      <ul>
-        <li>{props.reward.item1}</li>
-        <li>{props.reward.item2}</li>
-        <li>{props.reward.item3}</li>
-      </ul>
-    </div>
-    <div className='estDeliv'>ESTIMATED DELIVERY {props.reward.estDeliv}</div>
-    <div className='shipsTo'>SHIPS TO {props.reward.shipsTo}</div>
-    <div className='backers'>{props.reward.backers} backers</div>
-  </div>
-);
+class RewardTier extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      view: 'default'
+    };
+
+    this.handleWidgetClick = this.handleWidgetClick.bind(this);
+    this.renderView = this.renderView.bind(this);
+  }
+
+  handleWidgetClick() {
+    this.setState({
+      view: 'clicked'
+    });
+  }
+
+  renderView() {
+    let {view} = this.state;
+    if (view === 'clicked') {
+      return (
+        <StyledMiniPledgeForm>
+          <MiniPledgeForm reward={this.props.reward}/>
+        </StyledMiniPledgeForm>
+      );
+    }
+  }
+
+  render() {
+    return (
+      <RewardWrapper>
+        <TierWrapper id={`${this.props.reward.id}`} className='rewardTier'>
+          <TitleWrapper className='pledgeAmount'>Pledge US$ {this.props.reward.pledgeAmount} or more</TitleWrapper>
+          <RewardName className='rewardName'>{this.props.reward.name}</RewardName>
+          <RewardDesc className='rewardDesc'>{this.props.reward.description}</RewardDesc>
+          <div className='rewardItems'>
+            <GenericWrapper>INCLUDES:</GenericWrapper>
+            <ul>
+              <ListWrapper>{this.props.reward.item1}</ListWrapper>
+              <ListWrapper>{this.props.reward.item2}</ListWrapper>
+              <ListWrapper>{this.props.reward.item3}</ListWrapper>
+            </ul>
```

Comment:
This seems to indicate that you are hard-coding three reward items into each mock campaign. Is that the case?

Reply:
Yes it is. Every reward tier has three reward items that it needs to display.